lib/eventboss/unit_of_work.rb

```diff
@@ -21,7 +21,17 @@ def run(client)
     rescue StandardError => exception
       handle_exception(exception, processor: processor, message_id: @message.message_id)
     else
-      cleanup(client)
+      cleanup(client) unless @postponed_by
+    ensure
+      change_message_visibility(client) if @postponed_by
+    end
+
+    def change_message_visibility(client)
+      client.change_message_visibility(
+        queue_url: @queue.url,
+        receipt_handle: @message.receipt_handle,
+        visibility_timeout: @listener.postponed_by
```

Comment:
Should this be?
```suggestion
        visibility_timeout: @postponed_by
```

Reply:
Well actually it should be
```ruby
@listener.postpone_by
```
everywhere

Reply:
Changed that